src/launch/lib/service/template/functions.py

```diff
@@ -0,0 +1,308 @@
+                dest_base=dest_base,
+                config=value,
+                parent_keys=current_keys,
+                skip_uuid=skip_uuid,
+                dry_run=dry_run,
+            )
+        else:
+            updated_config[key] = value
+
+    return updated_config
+
+
+def copy_template_files(
+    src_dir: Path, target_dir: Path, exclude_dir: str, dry_run: bool = True
+) -> None:
+    """
+    Copies files from a source directory to a target directory, excluding a specific directory.
+
+    Args:
+        src_dir (Path): The source directory containing the files to be copied.
+        target_dir (Path): The target directory where the files will be copied.
+        exclude_dir (str): The directory to exclude from the copy operation.
+
+    Returns:
+        None
+    """
+    if dry_run:
+        click.secho(
+            f"[DRYRUN] Processing template, would have copied files: {src_dir=} {target_dir=} {exclude_dir=}",
+            fg="yellow",
+        )
+        return
+    os.makedirs(target_dir, exist_ok=True)
+
+    for item in os.listdir(src_dir):
+        src_item = os.path.join(src_dir, item)
+        target_item = os.path.join(target_dir, item)
+        if os.path.isdir(src_item):
+            if item != exclude_dir and item != ".git":
+                shutil.copytree(src_item, target_item, dirs_exist_ok=True)
+        else:
+            shutil.copy2(src_item, target_item)
+
+
+def list_jinja_templates(base_dir: str) -> tuple:
+    base_path = Path(base_dir)
+    template_paths = []
+    modified_paths = []
+    pattern = re.compile(r"\{\{.*?\}\}")
+
+    for jinja_file in base_path.rglob("*.j2"):
+        modified_path = pattern.sub("*", str(jinja_file))
+        modified_path = modified_path.replace(str(base_path), "")
+        modified_path = modified_path.lstrip("/")
+        modified_paths.append(modified_path)
+        template_paths.append(jinja_file.as_posix())
+
+    return template_paths, modified_paths
+
+
+def render_jinja_template(
+    template_path: Path,
+    destination_dir: str,
+    file_name: str,
+    template_data: dict = {"data": None},
+    dry_run: bool = True,
+) -> None:
+    if not template_data.get("data"):
+        template_data["data"] = {}
+
+    env = Environment(loader=FileSystemLoader(template_path.parent))
+    template = env.get_template(template_path.name)
+    template_data["data"]["path"] = str(destination_dir)
+    template_data["data"]["config"]["dir_dict"] = get_value_by_path(
+        template_data["data"]["config"][PLATFORM_SRC_DIR_PATH],
+        str(destination_dir)[(str(destination_dir).find(PLATFORM_SRC_DIR_PATH) + 9) :],
+    )
+    output = template.render(template_data)
+    destination_path = destination_dir / file_name
+
+    if output.strip():
+        with open(destination_path, "w") as f:
+            if dry_run:
+                click.secho(
+                    f"[DRYRUN] Rending template, would have saved rendered file: {destination_path}",
```

Comment:
```suggestion
                    f"[DRYRUN] Rendering template, would have saved rendered file: {destination_path}",
```

There are a few other typos in this vein.